### PR TITLE
Version bumps BT-332

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

**Comment by cjllanwarne** [comment]
Just OOO, is this mostly complementary, orthogonal, or replacing-of, the scala steward update PRs?

**Comment by kshakir** [comment]
> Just OOO, is this mostly complementary, orthogonal, or replacing-of, the scala steward update PRs?

I haven't gone through the various other PRs yet to see what they're fixing or not. Over weekends I've been experimenting with updating various subsystems I'm already familiar with and seeing if Travis likes the changes. If I had to guess, there's probably a bit of overlap with the version bumps here and the scala steward PRs.

Things done here, and may or may not have been addressed in the stewarded PRs:
- Some non-semver versions have been updated/fixed. Does scala steward do date comparisons or only semver? (ex: nl.grons.metrics(3), apache commons, workbench-libs, etc.)
- Some intermediate version fixes have been applied. The versions listed in `develop` will be out of date, while the absolute latest available version may not be compatible (ex: cats-effect, fs2, http4s, etc.)
- Some SDKs had a few deprecated functions and required a little RTFMing 📖 (ex: sentry)

Btw, as it's not a blocker (yet) some weekend I or someone else will have to dive into statsd and deal with those libs plus whatever our bespoke statsd-proxy is doing... 🤔

File: project/Dependencies.scala
Comment (breilly2) [inline project/Dependencies.scala:13]: Will need to make sure mutect2 test passes post-merge.

File: project/Dependencies.scala
Comment (breilly2) [inline project/Dependencies.scala:4]: I cannot find any release notes/changelog/version history saying what has changed. Going to have to rely on tests here. :(

Comment (kshakir) [reply project/Dependencies.scala:4]: Here's [a TL;DR changelog](https://github.com/hseeberger/akka-http-json/compare/v1.35.3...v1.37.0)... that effectively points to other changelogs 🪀🐶

File: project/Dependencies.scala
Comment (breilly2) [inline project/Dependencies.scala:19]: I don't know if this will affect us, but it's described as silently source-breaking: https://github.com/typelevel/cats/pull/3777 (from 2.4.1-hotfix). Can someone else check on this?

Comment (kshakir) [reply project/Dependencies.scala:19]: I'm not sure if it will, but `Parallel` is used around CWL/WOM. So maybe? Although I've used a LOT of other parts of cats, I'm not super familiar with `Parallel`. I'm also not super familiar with Cromwell's language parsing, leaving me unsure if there would be negative tests to verify error accumulation works as expected. 🤔

File: project/Dependencies.scala
Comment (breilly2) [inline project/Dependencies.scala:48]: The bump to 2.x includes updating the dependency on `google-cloud-shared-config` with a breaking change: https://github.com/googleapis/java-bigquery/releases/tag/v2.0.0 but I can't tell what, if anything, that means for us.

File: project/Dependencies.scala
Comment (breilly2) [inline project/Dependencies.scala:49]: I wonder if we should switch over to this: https://github.com/googleapis/java-kms

File: project/Dependencies.scala
Comment (breilly2) [inline project/Dependencies.scala:51]: Claims to include a breaking change in 3.x: https://github.com/googleapis/java-monitoring/releases/tag/v3.0.0 but I'm going to guess that it won't affect us.

File: project/Dependencies.scala
Comment (kshakir) [reply project/Dependencies.scala:49]: I'm not very experienced with the multiple options of Google's Java APIs. But if it's easy enough to port over to the new-ish (2019) idiomatic KMS API and centaur tests still pass then Sounds Good To Me.

File: project/Dependencies.scala
Comment (mcovarr) [reply project/Dependencies.scala:13]: kicked off AWS build 104 to get Jenkins' opinion before merge

**Review by mcovarr — APPROVED** [review]
👍 assuming Jenkins is okay with the AWS changes

File: services/src/test/scala/cromwell/services/database/LiquibaseComparisonSpec.scala
Comment (kshakir) [reply services/src/test/scala/cromwell/services/database/LiquibaseComparisonSpec.scala:351]: The keys in the map are what HSQLDB calls the column. The values are what the other RDBMS calls the column. In this case MariaDB.

The way the matching works, at the start of the test an HSQLDB schema is created with Slick instead of Liquibase. Then one by one each RDBMS type is initialized, liquibased, and then the resulting schema is compared to the HSQLDB-Slick schema. Any mismatches in column type are an error. Except sometimes there are minor differences.

In this line HSQLDB has columns that are `ColumnType("BIGINT", Option(64))` while MariaDB doesn't return sizes for "BIGINT" so its equivalent column type is a `CoulmnType("BIGINT", None)`.

File: project/Dependencies.scala
Comment (kshakir) [reply project/Dependencies.scala:13]: > kicked off AWS build 104 to get Jenkins' opinion before merge

FYI [104](https://fc-jenkins.dsp-techops.broadinstitute.org/job/cromwell-cron-aws/104/console) died due to the agent running out of space: `java.io.IOException: No space left on device`

Long live [106](https://fc-jenkins.dsp-techops.broadinstitute.org/job/cromwell-cron-aws/106/)

Comment (kshakir) [reply project/Dependencies.scala:13]: Update: [106](https://fc-jenkins.dsp-techops.broadinstitute.org/job/cromwell-cron-aws/106/) passed

File: project/Dependencies.scala
Comment (breilly2) [inline project/Dependencies.scala:52]: Oof. This goes back to last August: https://github.com/broadinstitute/cromwell/pull/5708/files#diff-ad2642dc77b3679e4ad2c7d3b2f59a2dcf48c1bf5b93a558a4fa44828315a34cR35. Maybe this one required code changes?

File: project/Dependencies.scala
Comment (breilly2) [inline project/Dependencies.scala:60]: 🤷

File: project/Dependencies.scala
Comment (breilly2) [inline project/Dependencies.scala:71]: If this all works, I love getting rid of pinned versions like this.

File: project/Dependencies.scala
Comment (breilly2) [inline project/Dependencies.scala:88]: Some stuff about character sets: https://dev.mysql.com/doc/relnotes/connector-j/8.0/en/news-8-0-26.html#connector-j-8-0-26-feature. Anything to note here?

File: project/Dependencies.scala
Comment (breilly2) [inline project/Dependencies.scala:95]: Ugh. Their link to the "release and upgrade notes" (https://rdf4j.org/release-notes#3-0-0) for 3.0.0 doesn't go to a useful place anymore so I don't know what to look for. I also don't know what RDF is or how we might use it.

Comment (kshakir) [reply project/Dependencies.scala:95]: - RDF: https://en.wikipedia.org/wiki/Resource_Description_Framework
- It's one of the specs used by the CWL spec ([example](https://pypi.org/project/cwltool/#modeling-a-cwl-document-as-rdf))
- This repo only uses rdf4j transitively. However, there was a vulnerability detected in an earlier version, and the OWL API dependency (yet another W3C spec) we use was slow to update their dependencies.
- For the curious about these libs in general there are [more external links regarding dependencies here](https://github.com/broadinstitute/cromwell/blob/67/project/Dependencies.scala#L450-L463)

File: project/Dependencies.scala
Comment (breilly2) [inline project/Dependencies.scala:110]: This feels like a huge jump and I wouldn't even know where to start, so... just saying that I'm leaving this one alone and assuming that any problems would be caught by compilation, tests, or general operations.

File: project/Dependencies.scala
Comment (breilly2) [inline project/Dependencies.scala:125]: I care about this, but I cannot find any reliable documentation links about specs2-mock or any version history. I hope they didn't change anything that will cause any of our tests to pass when they should fail.

File: project/Dependencies.scala
Comment (breilly2) [inline project/Dependencies.scala:137]: `-SNAP` ⁉️ 😬

File: engine/src/test/scala/cromwell/engine/workflow/WorkflowDockerLookupActorSpec.scala
Comment (breilly2) [inline engine/src/test/scala/cromwell/engine/workflow/WorkflowDockerLookupActorSpec.scala:66]: Is `.tell` the new `!`?

File: filesystems/oss/src/main/scala/cromwell/filesystems/oss/batch/OssBatchIoCommand.scala
Comment (breilly2) [inline filesystems/oss/src/main/scala/cromwell/filesystems/oss/batch/OssBatchIoCommand.scala:61]: Does this not return `void` anymore? Should we be paying attention to what it does return?

File: filesystems/oss/src/test/scala/cromwell/filesystems/oss/nio/OssNioUtilSpec.scala
Comment (breilly2) [inline filesystems/oss/src/test/scala/cromwell/filesystems/oss/nio/OssNioUtilSpec.scala:101]: Same question as in `OssBatchIoCommand.scala` above.

File: server/src/test/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActorSpec.scala
Comment (breilly2) [inline server/src/test/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActorSpec.scala:30]: Why this change?

Comment (kshakir) [reply server/src/test/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActorSpec.scala:30]: This spec, like others, effectively runs its own entire Cromwell instance. When run on Travis it sometimes takes longer than 20 seconds for the lazy-loaded in-memory DB to start up, especially when other tests don't clean 100% clean up leading to GC pauses etc.

See the comment "Yes, this might be slow..." further down below in this file for more slightly more context.

File: services/src/test/scala/cromwell/services/database/LobSpec.scala
Comment (breilly2) [inline services/src/test/scala/cromwell/services/database/LobSpec.scala:38]: Was this just always incorrectly named?

File: filesystems/oss/src/main/scala/cromwell/filesystems/oss/batch/OssBatchIoCommand.scala
Comment (kshakir) [reply filesystems/oss/src/main/scala/cromwell/filesystems/oss/batch/OssBatchIoCommand.scala:61]: > Should we be paying attention to what it does return?

No. It returns a `VoidResult`: https://github.com/aliyun/aliyun-oss-java-sdk/blob/3.13.1/src/main/java/com/aliyun/oss/OSSClient.java#L665

File: services/src/test/scala/cromwell/services/database/LobSpec.scala
Comment (kshakir) [reply services/src/test/scala/cromwell/services/database/LobSpec.scala:38]: Not "always". At some point the body of the test was updated and the name of the test wasn't.

File: project/Dependencies.scala
Comment (kshakir) [reply project/Dependencies.scala:125]: I care too. The reason it's hard to find docs on `specs2-mock` is that it seems to be [End Of Life](https://github.com/etorreborre/specs2/commits/master/mock/). See the cromwell sibling PR that kills this dependency off 😉🔪💀

As for some sort of docs:
- 4.12.0 mentions specs2's `specs2-mock` extension: https://etorreborre.github.io/specs2/guide/SPECS2-4.12.0/org.specs2.guide.Installation.html#other-dependencies
- 5.0 (in RC) no longer mentions the extensions and instead refers to `scalamock`: https://etorreborre.github.io/specs2/guide/5.0.0-RC-01/org.specs2.guide.Installation.html#other-dependencies

File: project/Dependencies.scala
Comment (kshakir) [reply project/Dependencies.scala:137]: Desperate times... (and not waiting for releases) 🤷

**Review by breilly2 — APPROVED** [review]
Looks good, as far as I can tell. I've done what I can to validate version bumps, and I'm trusting that the code changes are largely fallout from that. I also don't know the newer sbt stuff, so I'm trusting that that's all correct. Definitely glad there are other eyes on reviewing this.

File: project/Dependencies.scala
Comment (kshakir) [reply project/Dependencies.scala:88]: Character encodings, like date-times, are definitely something to keep an eye out for.

In general anyone not using UTC time and 7-bit ASCII may run into issues 😦

That said, we generally hope that any egregious encoding bugs are caught in centaur. For the curious, bugs in the past have popped up when validating and storing values of length "255" in our liquibased tables. What unit is that: bytes, characters, encoded characters in bytes?? If my memory is correct, the issues manifested as strings suddenly being too long to store in some of the tables.

**Pull request merged**
